libs/stats/odc/stats/tasks.py

```diff
@@ -55,16 +58,35 @@ def sanitize(v):
         if isinstance(v, datetime):
             return v.isoformat()
         return v
+
     return transform_object_tree(sanitize, query)
 
 
+def download_from_s3(cache):
```

Comment:
also why is parameter called `cache` this function download any s3 object to filesystem.

Reply:
Agree, making this a generic method to download a file will be better. Or even better, using a method from somewhere else...

The boto download file function is already very simple, so wrapping it up is not necessary, I think: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-download-file.html

``` python
import boto3

s3 = boto3.client('s3')
s3.download_file('BUCKET_NAME', 'OBJECT_NAME', 'FILE_NAME')
```

Reply:
@alexgleith yes it is necessary to support common features across all s3 access methods

- reading without credentials
- reading with specific set of credentials
- any other configuration that differs from default

Those features are not there because I have nothing better to do, they were added in response to real user requirements.

The way `odc.aws` library works is that you setup (or extract from the cache) properly configured `s3` client (that step takes care of various credential needs). Then, you use that to perform various s3 operations, whether that means listing buckets or fetching small objects into memory or fetching large objects to disk. This is library code we need to put some extra effort in, to make it usable in more contexts than just this specific use-case.

Reply:
have changed this to url.